docs/codeql/ql-language-reference/types.rst

```diff
@@ -382,6 +384,68 @@ from ``OneTwoThree`` and ``int``.
    must :ref:`override <overriding-member-predicates>` those definitions to avoid ambiguity.
    :ref:`Super expressions <super>` are often useful in this situation.
 
+.. _final-extensions:
+
+Final extensions
+================
+
+A class can extend final types or final aliases of types. In that case, it inherits final
+versions of all the member predicates and fields of those supertypes.
+Member predicates that are inherited through final extensions cannot be overridden,
+but they can be shadowed.
+
+For example, extending the class from the :ref:`first example <defining-a-class>`:
+
+.. code-block:: ql
+
+    final class FinalOneTwoThree = OneTwoThree;
+    
+    class OneTwo extends FinalOneTwoThree {
```

Comment:
Would it make sense to call this `OneTwoFromFinal` to distinguish it from the `OneTwo` class in the `Overriding member predicates` section? Otherwise, I think we need to explicitly say that for this example we imagine `OneTwoThree` has been defined as in the first example, but `OneTwo` from the `Overriding member predicates` section is not in scope.

Reply:
With the second commit, I have renamed the class `OneTwoFinalExtension`. I think this is better than `OneTwoFinal`, which might mislead readers to think the class is itself final.